diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2f76264..6fe5433 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -356,7 +356,7 @@ INSTALL(TARGETS h2o
     LIBRARY DESTINATION lib)
 
 ADD_EXECUTABLE(setuidgid src/setuidgid.c)
-INSTALL(TARGETS setuidgid RUNTIME DESTINATION share/h2o)
+INSTALL(TARGETS setuidgid RUNTIME DESTINATION libexec/h2o)
 
 INSTALL(DIRECTORY include/ DESTINATION include FILES_MATCHING PATTERN "*.h")
 
diff --git a/lib/handler/configurator/fastcgi.c b/lib/handler/configurator/fastcgi.c
index 310b080..c8bf2c9 100644
--- a/lib/handler/configurator/fastcgi.c
+++ b/lib/handler/configurator/fastcgi.c
@@ -284,7 +284,7 @@ static int on_config_spawn(h2o_configurator_command_t *cmd, h2o_configurator_con
         argv[i++] = dirname;
         argv[i++] = "--";
         if (spawn_pw != NULL) {
-            argv[i++] = h2o_configurator_get_cmd_path("share/h2o/setuidgid");
+            argv[i++] = h2o_configurator_get_cmd_path("libexec/h2o/setuidgid");
             argv[i++] = spawn_pw->pw_name;
         }
         argv[i++] = "/bin/sh";
